uvicorn/server.py

```diff
@@ -238,7 +239,11 @@ async def shutdown(self, sockets=None):
         if not self.force_exit:
             await self.lifespan.shutdown()
 
-    def install_signal_handlers(self):
+    def install_signal_handlers(self) -> None:
+        if threading.current_thread() is threading.main_thread():
```

Comment:
```suggestion
        if threading.current_thread() is not threading.main_thread():
```

Reply:
🤦 Obviously! I guess I didn't actually take the chance to run this locally either. 😆